code/weapon/weapons.cpp

```diff
@@ -4101,7 +4106,27 @@ void find_homing_object_cmeasures_1(object *weapon_objp)
 
 				if (dist < cm_wip->cm_effective_rad)
 				{
-					float	chance;
+					float chance;
+
+					if (wp->cmeasure_ignore_list == nullptr) {
+						wp->cmeasure_ignore_list = new SCP_vector<int>;
+						wp->cmeasure_ignore_list->reserve(cm_wip->shots * 2); // guesstimate the number of CMs this weapon will see
```

Comment:
How about removing the reserve entirely? I'm starting to think that predicting the number of CMs seen by each missile is practically impossible, and adding conditionals just clutters things up.